astropy/table/sorted_array.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+
+class SortedArray(object):
+    '''
+    Implements a sorted array container using
+    a list of numpy arrays.
+
+    Parameters
+    ----------
+    lines : `Table`
```

Comment:
It's because the end result has to include the rows in their original positions. For example to create an index on the table
```
<Table length=3>
 col0 col1
int64 str1
----- ----
    3    a
    1    b
    2    c
```
we add a column corresponding to row numbers:
```
<Table length=3>
 col0 col1  col2
int64 str1 int64
----- ---- -----
    3    a     0
    1    b     1
    2    c     2
```
and sort so that the last column is the new index `sorted_data`:
```
In [16]: t[np.lexsort([t['col1'], t['col0']])]
Out[16]: 
<Table length=3>
 col0 col1  col2
int64 str1 int64
----- ---- -----
    1    b     1
    2    c     2
    3    a     0
```